pkg/operator/ceph/cluster/osd/update.go

```diff
@@ -126,7 +126,7 @@ func (c *updateConfig) updateExistingOSDs(errs *provisionErrors) {
 		}
 
 		// backward compatibility for old deployments
-		if osdInfo.DeviceClass == "" {
+		if osdInfo.DeviceClass == "" || osdInfo.DeviceClass == "None" {
```

Comment:
As mentioned in the huddle I think using `,omitempty"` might better solve this instead of checking for None. Also I cannot get a repro on why we would get `None`. https://go.dev/play/p/xKyIFkmrIKl

Reply:
I added the `,omitempty"` at these two places,
https://github.com/rook/rook/blob/2519018b264ccf7fc7799213ad14e7a420d792ec/pkg/operator/ceph/cluster/osd/osd.go#L99

https://github.com/rook/rook/blob/2519018b264ccf7fc7799213ad14e7a420d792ec/pkg/daemon/ceph/client/osd.go#L313

But still, see that `osdInfo.DeviceClass` is set as `None`

Reply:
Hum I'd be curious to trace back and really understand why the command will return None. That shouldn't be the case.

Reply:
@leseb, I traced it out `None` value is returned from the env. variable https://github.com/rook/rook/blob/bd962e3e8533cd40d7bb68ceaf274e113b83af5d/pkg/operator/ceph/cluster/osd/osd.go#L495-L496

Reply:
Thanks, weird it is not supposed to since it defaults to "".

Reply:
yes, I am trying to fix it.

Reply:
I wonder if this https://github.com/rook/rook/blob/master/pkg/operator/ceph/cluster/osd/osd.go#L99 is not missing `,omitempty"` too?

Reply:
Actually, this is the culprit https://github.com/rook/rook/blob/master/pkg/daemon/ceph/osd/volume.go#L85,

see: 

```
  [block]       /dev/ceph-a8ae9f01-a440-4a0e-8e4d-592d3bce3a9d/osd-block-058f4926-6ee7-4165-9a55-db5a1fd22d2f

      block device              /dev/ceph-a8ae9f01-a440-4a0e-8e4d-592d3bce3a9d/osd-block-058f4926-6ee7-4165-9a55-db5a1fd22d2f
      block uuid                cr4xnI-h1sx-4DT3-ZC2N-0kI9-RtZF-Nkr3q1
      cephx lockbox secret      
      cluster fsid              f20931eb-9336-4234-a5c7-b0b44ab8c07a
      cluster name              ceph
      crush device class        None
      encrypted                 0
      osd fsid                  058f4926-6ee7-4165-9a55-db5a1fd22d2f
      osd id                    2
      osdspec affinity          
      type                      block
      vdo                       0
      devices                   /dev/vdd
```

With `crush device class        None`. 

In the end, it's valid to check for `None` since we don't control this behavior!

Reply:
I have checked it while testing it doesn't make any change.

I see the `ROOK_OSD_DEVICE_CLASS` environment variable is returned as `{ROOK_OSD_DEVICE_CLASS None nil}`

Reply:
> 

Thanks, got it.
This is because of how `ceph` returns the output `ceph-volume`.

Reply:
Please leave a comment in the code to explain why we need to check against `None`and add a link to this tracker: https://tracker.ceph.com/issues/53425